src/main/java/Duke.java

```diff
@@ -1,10 +1,176 @@
+import tasks.*;
+import java.util.Scanner;
+import java.util.*;
+import java.io.*;
+
 public class Duke {
-    public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+
+    private Storage storage;
+    private ArrayList<Task> allTasks = new ArrayList<Task>();
+    private Ui ui;
+
+    /**
+     *
+     * @param filePath
+     */
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+
+        try {
+            File dataDir = new File("./data");
+            File dataText = new File("./data/duke.txt");
+            if (!dataDir.exists()){
+                dataDir.mkdirs();
+            }
+            if (!dataText.exists()){
+                dataText.createNewFile();
+            }
+            allTasks = new ArrayList<Task>(storage.load());
+        } catch (IOException e) {
+            ui.showLoadingError();
+            allTasks = new ArrayList<Task>();
+
+        }
+    }
+
+    public void run() throws IOException {
+        boolean notQuit = true;
+        int index;
+        String description, detail, by, at;
+        Task t;
+
+        ui.greeting();
+
+        while (notQuit) {
+            String command = ui.readCommand().replaceAll("( )+", " ");
```

Comment:
Consider refactoring this into another function that parses command to make the code SLAP